awx/ui/client/src/templates/job_templates/edit-job-template/job-template-edit.controller.js

```diff
@@ -184,33 +185,71 @@ export default
             function jobTemplateLoadFinished(){
                 CreateSelect2({
                     element:'#job_template_job_type',
-                    multiple: false
+                    multiple: false,
+                    callback: 'select2Loaded',
+                    scope: $scope
                 });
 
                 CreateSelect2({
                     element:'#playbook-select',
-                    multiple: false
+                    multiple: false,
+                    callback: 'select2Loaded',
+                    scope: $scope
                 });
 
                 CreateSelect2({
                     element:'#job_template_job_tags',
                     multiple: true,
-                    addNew: true
+                    addNew: true,
+                    callback: 'select2Loaded',
+                    scope: $scope
                 });
 
                 CreateSelect2({
                     element:'#job_template_skip_tags',
                     multiple: true,
-                    addNew: true
+                    addNew: true,
+                    callback: 'select2Loaded',
+                    scope: $scope
                 });
 
                 CreateSelect2({
                     element: '#job_template_custom_virtualenv',
                     multiple: false,
-                    opts: $scope.custom_virtualenvs_options
+                    opts: $scope.custom_virtualenvs_options,
+                    callback: 'select2Loaded',
+                    scope: $scope
                 });
             }
 
+            $scope.$on('select2Loaded', () => {
+                select2Count++;
+                if (select2Count === 10) {
```

Comment:
Is there a way to derive this value so that it isn't hard-coded?

Reply:
What if we turned CreateSelect2 into something that returned a promise instead of emitting a callback.  We could resolve here: https://github.com/ansible/awx/pull/3260/files#diff-fc0666ec5ec9b4df38e3216cb69a3aafR708.  Then we could just wait for all the promises to resolve before setting the form to pristine and setting up the watcher.